lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1244,19 +1244,58 @@ def has_dict_changed(new_dict, old_dict):
     return False
 
 
-def has_list_changed(new_list, old_list):
+def has_list_changed(new_list, old_list, sort_lists=True, sort_key=None):
     """
-    Check two lists has differences.
+    Check two lists have differences. Sort lists by default.
     """
+
+    def sort_list(unsorted_list):
+        """
+        Sort a given list.
+        The list may contain dictionaries, so use the sort key to handle them.
+        """
+
+        try:
+            return sorted(unsorted_list)
+        except TypeError:
+            if isinstance(unsorted_list[0], dict):
+                if not sort_key:
+                    raise Exception(
+                        "A sort key was not specified when sorting list"
```

Comment:
We should try and use single-quotes. There's some inconsistencies in this module but the majority of strings are single-quoted.

Reply:
Done.